Sources/TuistServer/Client/ServerAuthenticationController.swift

```diff
@@ -42,75 +44,93 @@ public enum AuthenticationToken: CustomStringConvertible, Equatable {
-                credentials = try await credentialsStore.read(serverURL: URL(string: "https://cloud.tuist.io")!)
-            }
-            return try credentials.map {
-                if let refreshToken = $0.refreshToken {
-                    return .user(
-                        legacyToken: nil,
-                        accessToken: try $0.accessToken.map(parseJWT),
-                        refreshToken: try parseJWT(refreshToken)
-                    )
-                } else {
+        #if canImport(TuistSupport)
+            if ciChecker.isCI() {
+                if let configToken = environment.tuistVariables[Constants.EnvironmentVariables.token] {
+                    return .project(configToken)
+                } else if let deprecatedToken = environment.tuistVariables[Constants.EnvironmentVariables.deprecatedToken] {
                     ServiceContext.current?.logger?
-                        .warning("You are using a deprecated user token. Please, reauthenticate by running 'tuist auth login'.")
-                    return .user(
-                        legacyToken: $0.token,
-                        accessToken: nil,
-                        refreshToken: nil
-                    )
+                        .warning(
+                            "Use `TUIST_CONFIG_TOKEN` environment variable instead of `TUIST_CONFIG_CLOUD_TOKEN` to authenticate on the CI"
+                        )
+                    return .project(deprecatedToken)
+                } else {
+                    return nil
+                }
+            } else {
+                var credentials: ServerCredentials? = try await credentialsStore.read(serverURL: serverURL)
+                if isTuistDevURL(serverURL), credentials == nil {
+                    credentials = try await credentialsStore.read(serverURL: URL(string: "https://cloud.tuist.io")!)
+                }
+                return try credentials.map {
+                    if let refreshToken = $0.refreshToken {
+                        return .user(
+                            legacyToken: nil,
+                            accessToken: try $0.accessToken.map(parseJWT),
+                            refreshToken: try parseJWT(refreshToken)
+                        )
+                    } else {
+                        ServiceContext.current?.logger?
+                            .warning(
+                                "You are using a deprecated user token. Please, reauthenticate by running 'tuist auth login'."
+                            )
+                        return .user(
+                            legacyToken: $0.token,
+                            accessToken: nil,
+                            refreshToken: nil
+                        )
+                    }
                 }
             }
-        }
+        #else
+            return .user(
+                legacyToken: nil,
+                accessToken: JWT(
+                    token: "INSERT_HERE",
```

Comment:
What's this `INSERT_HERE` for?

Reply:
This is where you place your token that you can take from the API docs since the authentication is not implemented, yet